tests/framework/utils.py

```diff
@@ -60,29 +61,116 @@ class CONST:
 
-class KubernetesClusterManager:
+    @abstractmethod
+    def initialize_cluster(self, kind_config=None) -> None:
+        pass
+
+    @abstractmethod
+    def cleanup(self) -> None:
+        pass
+
+    @abstractmethod
+    def upload_image():
+        pass
+
+    @abstractmethod
+    def check_cluster_exist(self) -> bool:
+        pass
+    
+    @classmethod
+    def instance(cls):
+        if cls.EXTERNAL_CLUSTER in os.environ:
+            return ExternalClusterManager()
+        else:
+            return KindClusterManager()
+
+class ExternalClusterManager(ClusterManager):
+    def __init__(self) -> None:
+        config.load_kube_config()
+        self.k8s_client_dict = {}
+        self.k8s_client_dict.update(
+            {
+                CONST.K8S_V1_CLIENT_KEY: client.CoreV1Api(),
+                CONST.K8S_CR_CLIENT_KEY: client.CustomObjectsApi(),
+            }
+        )
+        self.cleanup_timeout = 120
+    
+    def cleanup(self, namespace = "default") -> None:
+        self.__delete_all_crs("ray.io", "v1alpha1", namespace, "rayservices")
+        self.__delete_all_crs("ray.io", "v1alpha1", namespace, "rayjobs")
+        self.__delete_all_crs("ray.io", "v1alpha1", namespace, "rayclusters")
+
+        k8s_v1_api = self.k8s_client_dict[CONST.K8S_V1_CLIENT_KEY]
+        start_time = time.time()
+        while time.time() - start_time < self.cleanup_timeout:
+            pods = k8s_v1_api.list_pod_for_all_namespaces(label_selector = 'app.kubernetes.io/created-by=kuberay-operator')
+            if len(pods.items) == 0:
+                logger.info("--- Cleanup rayservices, rayjobs, rayclusters %s seconds ---", time.time() - start_time)
+                break
+
+            time.sleep(1)
+
+        shell_subprocess_run("helm uninstall kuberay-operator", check=False)
+        start_time = time.time()
+        while time.time() - start_time < self.cleanup_timeout:
+            pods = k8s_v1_api.list_pod_for_all_namespaces(label_selector = 'app.kubernetes.io/component=kuberay-operator')
+            if len(pods.items) == 0:
+                logger.info("--- Cleanup kuberay-operator %s seconds ---", time.time() - start_time)
+                break
+
+            time.sleep(1)
+            
+        for _, k8s_client in self.k8s_client_dict.items():
+            k8s_client.api_client.rest_client.pool_manager.clear()
+            k8s_client.api_client.close()
+
+    def initialize_cluster(self, kind_config=None) -> None:
+        pass
+
+    def upload_image(self, image):
+        pass
+
+    def check_cluster_exist(self) -> bool:
+        """Check whether cluster exists or not"""
+        return (
+            shell_subprocess_run(
+                "kubectl cluster-info --context kind-kind", check=False
+            )
+            == 0
+        )
+    
+    def __delete_all_crs(self, group, version, namespace, plural):
+        custom_objects_api = self.k8s_client_dict[CONST.K8S_CR_CLIENT_KEY]
```

Comment:
The custom resource lifecycles are better controlled by `CREvent` ([link](https://github.com/ray-project/kuberay/blob/2292e61b9155d0cd94d11ef1af987f6ff3559eb7/tests/framework/prototype.py#L160-L212)). In my opinion, the cleanup at the cluster level is not necessary. We can remove this function.